### DISCUSSION TITLE
Issue on redirect page using middleware with NextAuth

### DISCUSSION THREAD

**Discussion by devpedro-dias** [community/community#137106] [Programming Help]
### Issue on Barber's View

I'm working on a barber shop website. My website supports 2 types of views, User and Barber. User view is complete and working properly, but Barber view is not. The problem is that when you click on "Dashboard" button, you are redirecting to "/". Following my project's business rules, only Barber can enter this page, I was looking for how to use middleware in docs, but I can't make it work. 

If anone can help, I would be very grateful.

Here is my repo: github.com/devpedro-dias/devbarber

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ebndev** [comment]
Hi @devpedro-dias, thanks for being a part of the GitHub Community!

You should check out this discussion: "[How to Secure Admin and Client Sides with Role-Based Redirection in Next.js + Express + MongoDB?](https://github.com/orgs/community/discussions/136655)"

You may be encountering a similar issue that can be resolved by the answer they posted there. If not, perhaps @Imadnajam can help you debug 😄

**Reply by devpedro-dias** [reply]
Thanks for the answer, after a while trying to solve the problem, I managed to do it. But if I have any questions related to it, I can check the link you indicated. Thanks!
